src/acceptance-test/java/org/zalando/nakadi/webservice/hila/HilaAT.java

```diff
@@ -73,6 +74,24 @@ public void before() throws IOException {
         this.subscription = createSubscription(subscription);
     }
 
+    @Test(timeout = 10000)
+    public void whenStreamTimeoutReachedPossibleToCommit() throws Exception {
+        publishEvent(eventType.getName(),"{\"foo\":\"bar\"}");
+        publishEvent(eventType.getName(),"{\"foo\":\"bar\"}");
+        final TestStreamingClient client = TestStreamingClient
+                .create(URL, subscription.getId(), "stream_limit=1&stream_timeout=1")
+                .start();
+        waitFor(() -> Assert.assertFalse(client.getBatches().isEmpty()), TimeUnit.SECONDS.toMillis(1), 100);
+        final SubscriptionCursor toCommit = client.getBatches().get(0).getCursor();
+        client.close(); // connection is closed, and stream as well
+
```

Comment:
Or maybe query for the metrics that keep track of the amount of consumer connected. When it reaches 0 then proceed.